sqlmesh/core/engine_adapter/snowflake.py

```diff
@@ -200,7 +204,9 @@ def _df_to_source_queries(
         batch_size: int,
         target_table: TableName,
     ) -> t.List[SourceQuery]:
-        temp_table = self._get_temp_table(target_table or "pandas")
+        temp_table = self._get_temp_table(
+            target_table or "pandas", quoted=False
```

Comment:
why do we need to unquote this? can we just make sure evyrtihng is quoted?

Reply:
Nope, because these get passed to `write_pandas()`. This function comes from the Snowflake library. It has a bunch of problems:

- It has a flag called `quote_identifiers` which defaults to `True`. So if you give it already quoted identifiers, it happily double quotes them.
- If you set `quote_identifiers=False`, then it doesnt quote the column names. Therefore, the table gets created with columns like `ID` but the rest of the SQLMesh code wants to query it as `"id"`, causing an error.

The path of least resistance was to simply not quote the identifiers first. This means that:
 - `write_pandas()` quotes them and creates correctly named tables with correctly named columns
 - `EngineAdapter.execute()` also quotes them when `create_table()` is called, because it see's they're unquoted when it goes to run the query